enterprise_gateway/services/external/k8s_client.py

```diff
@@ -0,0 +1,25 @@
+"""Contains factory to create kubernetes api client instances using a single confguration"""
+import os
+
+from kubernetes import client, config
+
+
+class KubernetesClientFactory:
+    """Manages kubernetes client creation from environment variables"""
+    def __init__(self):
```

Comment:
Nit: Constructors always return `None`

```suggestion

class KubernetesClientFactory:
    """Manages kubernetes client creation from environment variables"""
    def __init__(self) -> None:
```